File: JavaServletSrc/arrt_chkdiana_sched.java
import java.io.*;
import java.text.*;
import java.util.*;
import javax.servlet.*;
import javax.servlet.http.*;
import com.zucchetti.sitepainter.SPPrxycizer;

@com.zucchetti.sitepainter.EntityRoles({})
public class arrt_chkdiana_sched extends SPRoutineServlet implements SPInvokable {
  public static String m_cEntityUid = "!*5768649f*!";
  public class ServletStatus extends SPRoutineServletStatus {
    public double _codice;
    public String cRitorno;
    public String w_codinter;
    public String w_tipinter;
    public String w_codage;
    public String w_codcli;
    public String w_ragben;
    public String daope;
    public String a_ope;
    public String dareg;
    public String a_reg;
    public String w_tipoarchd;
    public String w_tipoarchp;
    public String w_stoper;
    public String w_flgope;
    public String w_flgfrz;
    public String w_flgrae;
    public String w_flgrap;
    public String w_tiprap;
    public String w_flgtit;
    public String w_tiptit;
    public String w_flgint;
    public String w_flgdel;
    public String w_flgchk;
    public String w_flgxls;
    public String w_flgsto;
    public String w_allsto;
    public String w_multifile;
    public String aaaabaaaaaaaaaaaaaaaaaaaaa_retval;
    public void GetNonLocals(SPParameterSource source) {
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_codinter","w_codinter");
      }        
      w_codinter = source.GetParameter("w_codinter","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_tipinter","w_tipinter");
      }        
      w_tipinter = source.GetParameter("w_tipinter","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_codage","w_codage");
      }        
      w_codage = source.GetParameter("w_codage","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_codcli","w_codcli");
      }        
      w_codcli = source.GetParameter("w_codcli","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_ragben","w_ragben");
      }        
      w_ragben = source.GetParameter("w_ragben","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.daope","daope");
      }        
      daope = source.GetParameter("daope","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.a_ope","a_ope");
      }        
      a_ope = source.GetParameter("a_ope","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.dareg","dareg");
      }        
      dareg = source.GetParameter("dareg","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.a_reg","a_reg");
      }        
      a_reg = source.GetParameter("a_reg","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_tipoarchd","w_tipoarchd");
      }        
      w_tipoarchd = source.GetParameter("w_tipoarchd","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_tipoarchp","w_tipoarchp");
      }        
      w_tipoarchp = source.GetParameter("w_tipoarchp","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_stoper","w_stoper");
      }        
      w_stoper = source.GetParameter("w_stoper","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_flgope","w_flgope");
      }        
      w_flgope = source.GetParameter("w_flgope","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_flgfrz","w_flgfrz");
      }        
      w_flgfrz = source.GetParameter("w_flgfrz","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_flgrae","w_flgrae");
      }        
      w_flgrae = source.GetParameter("w_flgrae","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_flgrap","w_flgrap");
      }        
      w_flgrap = source.GetParameter("w_flgrap","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_tiprap","w_tiprap");
      }        
      w_tiprap = source.GetParameter("w_tiprap","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_flgtit","w_flgtit");
      }        
      w_flgtit = source.GetParameter("w_flgtit","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_tiptit","w_tiptit");
      }        
      w_tiptit = source.GetParameter("w_tiptit","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_flgint","w_flgint");
      }        
      w_flgint = source.GetParameter("w_flgint","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_flgdel","w_flgdel");
      }        
      w_flgdel = source.GetParameter("w_flgdel","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_flgchk","w_flgchk");
      }        
      w_flgchk = source.GetParameter("w_flgchk","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_flgxls","w_flgxls");
      }        
      w_flgxls = source.GetParameter("w_flgxls","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_flgsto","w_flgsto");
      }        
      w_flgsto = source.GetParameter("w_flgsto","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_allsto","w_allsto");
      }        
      w_allsto = source.GetParameter("w_allsto","");
      if (source instanceof SPRequestSource) {
       ((SPRequestSource)source).request.setAttribute("com.zucchetti.sitepainter.parameters.w_multifile","w_multifile");
      }        
      w_multifile = source.GetParameter("w_multifile","");
    }
  }
  public SPEntityGlobals getEntityGlobals() {
    // Costanti globali dell' entita'
    SPRoutineGlobals l_eg;
    l_eg = new SPRoutineGlobals();
    l_eg.name = "arrt_chkdiana_sched";
    l_eg.hasProgressItem = false;
    l_eg.isPublic = false;
    return l_eg;
  }
  public void doProcess(HttpServletRequest request,HttpServletResponse response) throws IOException, ServletException {
    // Inizializzazione delle variabili della classe globale di stato
    ServletStatus status;
    status = new ServletStatus();
    status.containing = this;
    SPParameterSource source;
    source = SPLib.GetSource(request);
    if ( ! (EnterServlet(status,source,request,response))) {
      return;
    }
    Begin(status);
  }
  boolean NeedsFetchPage(ServletStatus status) {
    return false;
  }
  private void ExecuteRoutine(ServletStatus status) throws IOException {
    arrt_chkdiana_schedR Rcalled;
    Rcalled = new arrt_chkdiana_schedR(status.context,status);
    if (CPLib.ne(status.m_cAction,"executeAsync")) {
      ExecuteWorkflowStartingScript(status,Rcalled);
      status.aaaabaaaaaaaaaaaaaaaaaaaaa_retval = Rcalled.Run(status.w_codinter,status.w_tipinter,status.w_codage,status.w_codcli,status.w_ragben,status.daope,status.a_ope,status.dareg,status.a_reg,status.w_tipoarchd,status.w_tipoarchp,status.w_stoper,status.w_flgope,status.w_flgfrz,status.w_flgrae,status.w_flgrap,status.w_tiprap,status.w_flgtit,status.w_tiptit,status.w_flgint,status.w_flgdel,status.w_flgchk,status.w_flgxls,status.w_flgsto,status.w_allsto,status.w_multifile);
      if (Rcalled.m_bError) {
        status.m_bError = true;
        status.m_cErrMsg = Rcalled.m_cLastMsgError;
      } else if ( ! (CPLib.Empty(status.m_cFaultTrace))) {
        status.m_bError = true;
        status.m_cErrMsg = status.m_cFaultTrace;
      }
      if (CPLib.eqr(CPLib.Left(status.m_cAtExit,8),"workflow")) {
        ExitToWorkflow(status);
      }
      if (Forward.IsForwarded(status.forwardedto)) {
        if (CPLib.IsNull(status.forwardedto.GetParameterString("m_cAtExit",null))) {
          status.forwardedto.SetParameter("m_cAtExit",status.m_cAtExit);
        }
        SPLib.ForwardTo(status.forwardedto,servletContext,status.request,status.response);
        return;
      }
    }
    BringBackPage(status,Rcalled);
  }
  CallerWithObjs NewRoutine(SPServletStatus status,Caller dataFromClient) {
    return new arrt_chkdiana_schedR(status.context,dataFromClient);
  }
  protected void BringBackPage(ServletStatus status,arrt_chkdiana_schedR Rcalled) throws IOException {
    String l_cErrMsg;
    String msg;
    msg = status.context.Translate("MSG_END_ROUTINE");
    SPLib.SetContentType(status.response);
    SPLib.NoCache(status.response);
    status.out = status.response.getWriter();
    String l_cJsUid;
    status.out.println(
     "<!doctype html>" +
    "");
    status.out.println(
     "<html>" +
    "");
    status.out.println(
     "<head>" +
    "");
    status.out.println(
     "<title>" +
     ""+status.context.Translate("MSG_ROUTINE_WIND")+"" +
     "</title>" +
    "");
    l_cJsUid = SPPrxycizer.proxycizedPath("stdFunctions.js");
    status.out.println(
     "<script src='../"+l_cJsUid+"' type='text/javascript'>" +
     "</script>" +
    "");
    l_cJsUid = SPPrxycizer.proxycizedPath("controls.js");
    status.out.println(
     "<script src='../"+l_cJsUid+"' type='text/javascript'>" +
     "</script>" +
    "");
    status.out.println(
     "<link type='text/css' rel='STYLESHEET' href='../"+status.m_cTheme+"/formPage.css'>" +
    "");
    status.out.println(
     "</head>" +
    "");
    if (status.m_bError) {
      msg = status.context.Translate("MSG_END_ROUTINE_WITH_MSG");
      status.out.println(
       "<body onkeydown='window.ManageKeys(event,false,false)'>" +
      "");
    } else {
      status.out.println(
       "<body>" +
      "");
    }
    status.out.println(
     "<p class='TabLabel' style='margin:0'>" +
     ""+msg+"" +
     "</p>" +
    "");
    if (status.m_bError) {
      status.out.println(
       "<script type='text/javascript'>" +
      "");
      l_cErrMsg = status.context.Translate(com.zucchetti.sitepainter.Library.explicativeErrorMessage(status.m_cErrMsg));
      status.out.println("detailed_database_error="+SPLib.ToJSValue(l_cErrMsg,"C",0,0)+";");
      status.out.println("m_cBrowser=" + SPLib.ToJSValue(status.m_cBrowser,"C",0,0)+ ";");
      status.out.println(
       "</script>" +
      "");
    }
    status.out.println(
     "<!--" +
    "");
    if ( ! (CPLib.Empty(Rcalled.m_cLastMsgError))) {
      CPTokenSource ts;
      ts = new CPTokenSource(status.context.Translate(com.zucchetti.sitepainter.Library.explicativeErrorMessage(Rcalled.m_cLastMsgError)),"\n");
      while (ts.HasMoreTokens()) {
        status.out.println("Error message:"+com.zucchetti.sitepainter.Library.ToHTMLEuro(ts.NextToken())+"");
      }
    }
    if ( ! (CPLib.Empty(status.m_cFaultTrace))) {
      status.out.println("Fault:["+SPLib.ToJSValue(status.context.Translate("MSG_ROUTINE_FAULTED","arrt_chkdiana_sched"),"C",0,0)+","+SPLib.ToJSValue(status.m_cFaultTrace,"C",0,0)+"]");
      if (CPLib.Empty(Rcalled.m_cLastMsgError) &&  ! (status.m_bApplet)) {
        status.out.println("Error message:"+com.zucchetti.sitepainter.Library.ToHTMLEuro(status.m_cFaultTrace)+"");
      }
    }
    status.PrintMutableParameters();
    status.out.println("Function return value:"+com.zucchetti.sitepainter.Library.ToHTMLEuro(CPLib.ToProperties(status.aaaabaaaaaaaaaaaaaaaaaaaaa_retval))+"");
    status.out.println(
     " -->" +
    "");
    status.out.println(
     "</body>" +
    "");
    status.out.println(
     "</html>" +
    "");
  }
  private void Begin(ServletStatus status) throws IOException {
    if ( ! (CPLib.Empty(status.m_cGlobalsToRead))) {
      JSGlobals(status);
    } else if (CPLib.Empty(status.m_cBrowseName)) {
      ExecuteRoutine(status);
    } else {
      JSResultSet(status);
    }
  }
  void IncludeJS(ServletStatus status) {
    /* --- Area Manuale = UI - Include */
    /* --- Fine Area Manuale */
  }
  // m_CallableNames inizializzato dal primo accesso a GetCallableNames
  private static volatile String[] m_CallableNames;
  public static String[] GetCallableNames() {
    if (CPLib.IsNull(m_CallableNames)) {
      java.util.Set building = new java.util.HashSet();
      GetCallableNames(building);
      m_CallableNames = CPLib.SetOfStringsAsArray(building);
    }
    return (String[])m_CallableNames.clone();
  }
  public static void GetCallableNames(java.util.Set callable) {
  }
  String[] RoutineIDChain() {
    return GetCallableNames();
  }
}

File: JavaServletSrc/arrt_chkdiana_sched.js
function arrt_chkdiana_sched() {
var w_codinter;
var w_tipinter;
var w_codage;
var w_codcli;
var w_ragben;
var daope;
var a_ope;
var dareg;
var a_reg;
var w_tipoarchd;
var w_tipoarchp;
var w_stoper;
var w_flgope;
var w_flgfrz;
var w_flgrae;
var w_flgrap;
var w_tiprap;
var w_flgtit;
var w_tiptit;
var w_flgint;
var w_flgdel;
var w_flgchk;
var w_flgxls;
var w_flgsto;
var w_allsto;
var w_multifile;
var m_Caller
arrt_chkdiana_sched._p=_rargs(arguments,{eval:function(e){eval(e)}})
function i_BlankParameters() {
w_codinter='';
w_tipinter='';
w_codage='';
w_codcli='';
w_ragben='';
daope='';
a_ope='';
dareg='';
a_reg='';
w_tipoarchd='';
w_tipoarchp='';
w_stoper='';
w_flgope='';
w_flgfrz='';
w_flgrae='';
w_flgrap='';
w_tiprap='';
w_flgtit='';
w_tiptit='';
w_flgint='';
w_flgdel='';
w_flgchk='';
w_flgxls='';
w_flgsto='';
w_allsto='';
w_multifile='';
}
if(arrt_chkdiana_sched._p.length>0)_rargs([[['w_codinter',arrt_chkdiana_sched._p[0]],['w_tipinter',arrt_chkdiana_sched._p[1]],['w_codage',arrt_chkdiana_sched._p[2]],['w_codcli',arrt_chkdiana_sched._p[3]],['w_ragben',arrt_chkdiana_sched._p[4]],['daope',arrt_chkdiana_sched._p[5]],['a_ope',arrt_chkdiana_sched._p[6]],['dareg',arrt_chkdiana_sched._p[7]],['a_reg',arrt_chkdiana_sched._p[8]],['w_tipoarchd',arrt_chkdiana_sched._p[9]],['w_tipoarchp',arrt_chkdiana_sched._p[10]],['w_stoper',arrt_chkdiana_sched._p[11]],['w_flgope',arrt_chkdiana_sched._p[12]],['w_flgfrz',arrt_chkdiana_sched._p[13]],['w_flgrae',arrt_chkdiana_sched._p[14]],['w_flgrap',arrt_chkdiana_sched._p[15]],['w_tiprap',arrt_chkdiana_sched._p[16]],['w_flgtit',arrt_chkdiana_sched._p[17]],['w_tiptit',arrt_chkdiana_sched._p[18]],['w_flgint',arrt_chkdiana_sched._p[19]],['w_flgdel',arrt_chkdiana_sched._p[20]],['w_flgchk',arrt_chkdiana_sched._p[21]],['w_flgxls',arrt_chkdiana_sched._p[22]],['w_flgsto',arrt_chkdiana_sched._p[23]],['w_allsto',arrt_chkdiana_sched._p[24]],['w_multifile',arrt_chkdiana_sched._p[25]]]],{eval:function(e){eval(e)}})
arrt_chkdiana_sched._s=new BatchJavascript();
if (Ne(typeof(document.FSender),'undefined') && Ne(typeof(document.FSender.m_cRegionalSettings),'undefined')) {
arrt_chkdiana_sched._s.SetParameterString('m_cRegionalSettings',document.FSender.m_cRegionalSettings.value);
}
arrt_chkdiana_sched._s.SetParameterString('w_codinter',WtA(w_codinter,'C'));
arrt_chkdiana_sched._s.SetParameterString('w_tipinter',WtA(w_tipinter,'C'));
arrt_chkdiana_sched._s.SetParameterString('w_codage',WtA(w_codage,'C'));
arrt_chkdiana_sched._s.SetParameterString('w_codcli',WtA(w_codcli,'C'));
arrt_chkdiana_sched._s.SetParameterString('w_ragben',WtA(w_ragben,'C'));
arrt_chkdiana_sched._s.SetParameterString('daope',WtA(daope,'C'));
arrt_chkdiana_sched._s.SetParameterString('a_ope',WtA(a_ope,'C'));
arrt_chkdiana_sched._s.SetParameterString('dareg',WtA(dareg,'C'));
arrt_chkdiana_sched._s.SetParameterString('a_reg',WtA(a_reg,'C'));
arrt_chkdiana_sched._s.SetParameterString('w_tipoarchd',WtA(w_tipoarchd,'C'));
arrt_chkdiana_sched._s.SetParameterString('w_tipoarchp',WtA(w_tipoarchp,'C'));
arrt_chkdiana_sched._s.SetParameterString('w_stoper',WtA(w_stoper,'C'));
arrt_chkdiana_sched._s.SetParameterString('w_flgope',WtA(w_flgope,'C'));
arrt_chkdiana_sched._s.SetParameterString('w_flgfrz',WtA(w_flgfrz,'C'));
arrt_chkdiana_sched._s.SetParameterString('w_flgrae',WtA(w_flgrae,'C'));
arrt_chkdiana_sched._s.SetParameterString('w_flgrap',WtA(w_flgrap,'C'));
arrt_chkdiana_sched._s.SetParameterString('w_tiprap',WtA(w_tiprap,'C'));
arrt_chkdiana_sched._s.SetParameterString('w_flgtit',WtA(w_flgtit,'C'));
arrt_chkdiana_sched._s.SetParameterString('w_tiptit',WtA(w_tiptit,'C'));
arrt_chkdiana_sched._s.SetParameterString('w_flgint',WtA(w_flgint,'C'));
arrt_chkdiana_sched._s.SetParameterString('w_flgdel',WtA(w_flgdel,'C'));
arrt_chkdiana_sched._s.SetParameterString('w_flgchk',WtA(w_flgchk,'C'));
arrt_chkdiana_sched._s.SetParameterString('w_flgxls',WtA(w_flgxls,'C'));
arrt_chkdiana_sched._s.SetParameterString('w_flgsto',WtA(w_flgsto,'C'));
arrt_chkdiana_sched._s.SetParameterString('w_allsto',WtA(w_allsto,'C'));
arrt_chkdiana_sched._s.SetParameterString('w_multifile',WtA(w_multifile,'C'));
arrt_chkdiana_sched._s.SetParameterString('m_bApplet','true');
arrt_chkdiana_sched._s.CallServlet('arrt_chkdiana_sched');
if (Eq(typeof(m_bCalculating),'undefined') ||  ! (m_bCalculating)) {
m_cErrorFromRoutine=arrt_chkdiana_sched._s.errmsg;
}
return arrt_chkdiana_sched._s.GetString();
}
arrt_chkdiana_sched.CtxLoad_ = function(){return []}
